vnext/ReactUWP/Views/ShadowNodeBase.cpp

```diff
@@ -112,4 +130,74 @@ void ShadowNodeBase::UpdateTransformPS()
   }
 }
 
+void ShadowNodeBase::UpdateHandledKeyboardEvents(KeyboardType keyboardType, folly::dynamic const& value)
+{
+  if (keyboardType == KeyboardType::KeyDown) {
+    m_handledKeyDownKeyboardEvents = KeyboardHelper::FromJS(value);
+  }
+  else
+  {
+    m_handledKeyUpKeyboardEvents = KeyboardHelper::FromJS(value);
+  }
+
+  UpdateKeyboardEventHooks(m_view);
+}
+
+void ShadowNodeBase::UpdateKeyboardEventHooks(XamlView const& xamlView)
+{
+  auto shouldHook = ShouldHookKeyboardEvent(m_view);
+  if (shouldHook != m_keyboardEventHooked)
+  {
+    if (shouldHook && m_view)
+    {
+      m_previewKeyboardEventHandler->hook(m_view);
+      m_keyboardEventHandler->hook(m_view);
+      m_keyboardEventHooked = true;
+    }
+    else
+    {
+      m_previewKeyboardEventHandler->unhook();
+      m_keyboardEventHandler->unhook();
+      m_keyboardEventHooked = false;
+    }
+  }
+}
+
+bool ShadowNodeBase::ShouldHookKeyboardEvent(XamlView const& xamlView)
+{
+  return (xamlView && (m_handledKeyDownKeyboardEvents.size() > 0 || m_handledKeyUpKeyboardEvents.size() > 0));
```

Comment:
>xamlView [](start = 10, length = 8)

similar question - why the null check?

Reply:
For safe check. This function is removed.
I modified them to assert in other places.